src/lib.rs

```diff
@@ -911,7 +911,7 @@ impl CertificateParams {
 			// Write extensions
 			// According to the spec in RFC 2986, even if attributes are empty we need the empty attribute tag
 			writer.next().write_tagged(Tag::context(0), |writer| {
-				if !subject_alt_names.is_empty() {
+				if !subject_alt_names.is_empty() || !custom_extensions.is_empty() {
```

Comment:
Given the commit message text about a test failure, should this commit be moved around to appear before some of the other commits in this PR?

Reply:
I made the test change + fix in the same commit so they would all build green. The commit message text describes a manually created intermediate step.